pkg/registry/server/server.go

```diff
@@ -315,6 +321,39 @@ func ResolveAndMergeConfigsAndInjectTest(configs Getter, resolver Resolver, reso
 					mergedConfig.Operator.Substitutions = append(mergedConfig.Operator.Substitutions, config.Operator.Substitutions...)
 				}
 			}
+			if config.CanonicalGoRepository != nil {
+				mergedConfig.CanonicalGoRepositoryList = append(mergedConfig.CanonicalGoRepositoryList, api.RefRepository{
+					Ref:        ref,
+					Repository: *config.CanonicalGoRepository,
+				})
+			}
+			for step, resources := range config.Resources {
+				if step == "*" { // * is special, and we will only pull from one config to merge, this is "last in wins" we may need to do better in the future
+					mergedConfig.Resources[step] = resources
+				} else {
+					stepWithRef := fmt.Sprintf("%s-%s", step, ref)
+					mergedConfig.Resources[stepWithRef] = resources
+				}
+			}
+			if len(config.Releases) > 0 && len(mergedConfig.Releases) == 0 {
+				// Since the release configs "should" be identical, we can just use the first one we come across
```

Comment:
This is by design, right?

Reply:
Yes, I am reasonably confident it will work as is. There is a chance we will have to implement smarter logic later though.